### ISSUE TITLE
ffi 1.9.11 fails on windows

### ISSUE THREAD

**Issue by snowman78** [ffi/ffi#506]
Running on windows get undefined method when running ffi 1.9.11.   Locking on 1.9.10 the issue goes away.

NoMethodError: undefined method `start_with?' for :crypt32:Symbol
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/ffi-1.9.11-x64-mingw32/lib/ffi/library.rb:126:in `rescue in block (2 levels) in ffi_lib'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/ffi-1.9.11-x64-mingw32/lib/ffi/library.rb:111:in `block (2 levels) in ffi_lib'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/ffi-1.9.11-x64-mingw32/lib/ffi/library.rb:110:in `each'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/ffi-1.9.11-x64-mingw32/lib/ffi/library.rb:110:in `block in ffi_lib'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/ffi-1.9.11-x64-mingw32/lib/ffi/library.rb:100:in `map'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/ffi-1.9.11-x64-mingw32/lib/ffi/library.rb:100:in `ffi_lib'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/rest-client-2.0.0-x64-mingw32/lib/restclient/windows/root_certs.rb:85:in `<class:RootCerts>'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/rest-client-2.0.0-x64-mingw32/lib/restclient/windows/root_certs.rb:12:in `<top (required)>'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/rest-client-2.0.0-x64-mingw32/lib/restclient/windows.rb:7:in `require_relative'
C:/RubyTools/Ruby21-x64/lib/ruby/gems/2.1.0/gems/rest-client-2.0.0-x64-mingw32/lib/restclient/windows.rb:7:in `<top (required)>'

**Comment by glennsarti** [comment]
Just been hit with this as well.   Issue is that we can add libraries as a symbol or string but the code at https://github.com/ffi/ffi/blob/master/lib/ffi/library.rb#L126 assumes a string and is throwing errors as a symbol

**Comment by jeremydyoung** [comment]
I've been running into this as well.  Looks like it has to do with changes in https://github.com/ffi/ffi/commit/6d107564a3d46a4f0413ff7e3812ae5e2a95f81d

Apparently I don't know github well enough to see which pull request that was part of.

**Comment by glennsarti** [comment]
There was no PR by the looks of it.  It was a straight commit.

**Comment by mwrock** [comment]
chef appveyor pipeline just got hit

**Comment by tduehr** [comment]
Fixed.

**Comment by jeremydyoung** [comment]
I've been spending almost an hour trying to dig through the code to provide a tested fix to this, but I figured I'd let you know that the change in 1.9.12 simply moves the error 2 lines down to the append operation:

```
unless libname.to_s.start_with?("/")
  path = ['/usr/lib/','/usr/local/lib/'].find do |pth|
    File.exist?(pth + libname)
  end
  if path
    libname = path + libname
    retry
  end
end
```

```
             .../gems/ffi-1.9.12-x86-mingw32/lib/ffi/library.rb:128:in `+': no implicit conversion of Symbol into String (TypeError)
        from .../gems/ffi-1.9.12-x86-mingw32/lib/ffi/library.rb:128:in `block (3 levels) in ffi_lib'
        from .../gems/ffi-1.9.12-x86-mingw32/lib/ffi/library.rb:127:in `each'
        from .../gems/ffi-1.9.12-x86-mingw32/lib/ffi/library.rb:127:in `find'
        from .../gems/ffi-1.9.12-x86-mingw32/lib/ffi/library.rb:127:in `rescue in block (2 levels) in ffi_lib'
```
Sorry that I haven't provided a fix yet, not a wiz at Ruby...

**Comment by jeremydyoung** [comment]
I have manually tested the fix in https://github.com/ffi/ffi/pull/508 by @mwrock and it works for the problems I'm facing.

**Comment by jeremydyoung** [comment]
Subsequent issue opened that shows the problem with 1.9.12:  https://github.com/ffi/ffi/issues/510

**Comment by glennsarti** [comment]
I've also manually verified that the fix in #508 resolves my issue.
